// plugin-welcome/src/api/models/ControllersTreatment.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersTreatment
 */
export interface ControllersTreatment {
    /**
     * 
     * @type {string}
     * @memberof ControllersTreatment
     */
    added?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersTreatment
     */
    patientid?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersTreatment
     */
    status?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersTreatment
     */
    symptom?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersTreatment
     */
    userid?: number;
}

export function ControllersTreatmentFromJSON(json: any): ControllersTreatment {
    return ControllersTreatmentFromJSONTyped(json, false);
}

export function ControllersTreatmentFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersTreatment {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'added': !exists(json, 'added') ? undefined : json['added'],
        'patientid': !exists(json, 'patientid') ? undefined : json['patientid'],
        'status': !exists(json, 'status') ? undefined : json['status'],
        'symptom': !exists(json, 'symptom') ? undefined : json['symptom'],
        'userid': !exists(json, 'userid') ? undefined : json['userid'],
    };
}

export function ControllersTreatmentToJSON(value?: ControllersTreatment | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added': value.added,
        'patientid': value.patientid,
        'status': value.status,
        'symptom': value.symptom,
        'userid': value.userid,
    };
}
